nextjs/src/components/ClientUpload.tsx

```diff
@@ -27,14 +34,27 @@ export function ClientUpload() {
           loaded: 0,
           total: file.size,
           status: "uploading",
+          isMultipart,
         },
       }));
 
       try {
-        const { data } = await upload(`${file.name}`, file, {
+        // Use upload function with multipart option for large files
+        const result = await upload(`${file.name}`, file, {
           url: "/api/upload",
           access: "private",
-          onUploadProgress: ({ loaded, total, percentage }) => {
+          multipart: isMultipart,
+          partSize: isMultipart ? 1 * 1024 * 1024 : undefined, // 10MB parts for multipart
```

Comment:
I don't think this will be relevant, but having 10Mi parts means that there's a total filesize limit of 100Gi per file (S3API has a limit of 10,000 parts per multipart upload).